types/src/index.ts

```diff
@@ -0,0 +1 @@
+export * from '@server/types'
```

Comment:
Why this file is not in `types/src/server/index.ts`?

Reply:
This might be a subjective design decision but this export server typing by default 
> see [GUIDE.md udpates](https://github.com/larriereguichet/PeerTube/pull/1/files#diff-0305bd639760eb03793a66c13c02b3d19c13c6d005eb72c61d161ff245520b5b) for an illustration.

You might prefer to expose something else by default, maybe `shared` typings?

Reply:
We could also reexport all of `@server/types` `@server/models` and `@shared` here.
Just as a reminder this is also to avoid `@peertube/types/server/types` kind of imports

Reply:
okay no problem, let's start with only server typings first

Reply:
I did some testing by publishing and consuming [@larriereguichet/peertube-types](https://www.npmjs.com/package/@larriereguichet/peertube-types) in real life and actually... it doesn't work as I think it was and I'm disappointed :sob: to say the least ...

All the efforts I put into this to get nice little imports on the plugins side aren't working after all. It was working via `yarn link` but doesn't with the published package. I suppose we'll have to deal with not very elegant imports as a starter. 

I spend some more time to see if I could tackle at a builder level with `webpack` and `tsconfig-paths-webpack-plugin` it seems there are still issues with Typescript project reference that I couldn't figure out... :anger: 

I updated doc to reflect this.